pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixServerStarter.java

```diff
@@ -154,6 +161,36 @@ public HelixServerStarter(String helixClusterName, String zkAddress, PinotConfig
             Server.DEFAULT_CURRENT_DATA_TABLE_VERSION));
   }
 
+  /**
+   *  Invoke pinot environment provider factory's init method to register the environment provider &
+   *  fetch the overridden server configs for the invoked environment provider.
+   */
+  @Nullable
+  private Map<String, String> fetchEnvironmentProperties() {
+    PinotConfiguration environmentProviderConfigs = _serverConf.subset(
+        Server.PREFIX_OF_CONFIG_OF_ENVIRONMENT_PROVIDER_FACTORY);
+
+    if (environmentProviderConfigs.toMap().isEmpty()) {
+      LOGGER.info("No environment provider config values provided for server property: {}",
+          Server.PREFIX_OF_CONFIG_OF_ENVIRONMENT_PROVIDER_FACTORY);
+      return null;
```

Comment:
A null return type for a map leads to 3 states: null/empty/non-empty. I see you are handling the null case, but returning an empty map would imply the same thing? i.e. there are no overrides?